ffi/src/lib.rs

```diff
@@ -388,14 +402,98 @@ fn create_client_internal(
             let redis_error = err.into();
             errors::error_message(&redis_error)
         })?;
+
+    // Wrap the runtime in an Arc so it can be shared
+    let runtime = Arc::new(runtime);
+
     let client = runtime
-        .block_on(GlideClient::new(ConnectionRequest::from(request), None))
+        .block_on(GlideClient::new(
+            ConnectionRequest::from(request),
+            Some(push_tx),
+        ))
         .map_err(|err| err.to_string())?;
+
+    // Create the client adapter that will be returned and used as conn_ptr
     let core = Arc::new(CommandExecutionCore {
-        client,
-        client_type,
+        client: client.clone(),
+        client_type: client_type.clone(),
+    });
+    let client_adapter = Arc::new(ClientAdapter {
+        runtime: runtime.clone(),
+        core,
     });
-    Ok(ClientAdapter { runtime, core })
+    // Clone client_adapter before moving it into the async block
+    let client_adapter_clone = Arc::into_raw(client_adapter.clone()).addr();
+
+    // Check if the pubsub_callback function pointer is not null
+    // We can't directly use is_null() on function pointers, so we need to compare with std::ptr::null()
+    let null_fn: PubSubCallback = unsafe { std::mem::transmute(std::ptr::null::<()>()) };
+    let is_callback_valid = !std::ptr::eq(
+        pubsub_callback as *const () as *const u8,
+        null_fn as *const () as *const u8,
+    );
```

Comment:
If you do cast (`as`) you don't need transmute and vice versa. I think you can simplify the check to
```rs
if pubsub_callback as usize != 0 {
  ...
```

Reply:
done